drivers/Aqara/aqara-presence-sensor/src/fp2/device_manager.lua

```diff
@@ -0,0 +1,272 @@
+  elseif val == 6 then
+    device:emit_event(MovementSensor.movement("approaching"))
+  elseif val == 7 then
+    device:emit_event(MovementSensor.movement("movingAway"))
+  end
+end
+
+function device_manager.zone_presence_handler(driver, device, zone, evt_value)
+  if not device:supports_capability(capabilities.multipleZonePresence) then return end
+
+  local zoneInfo = multipleZonePresence.findZoneById(zone)
+  if not zoneInfo then
+    multipleZonePresence.createZone("zone" .. zone, zone)
+  end
+  local evt_action = multipleZonePresence.notPresent
+  if evt_value == 1 then evt_action = multipleZonePresence.present end
+  multipleZonePresence.changeState(zone, evt_action)
+  multipleZonePresence.updateAttribute(driver, device)
+end
+
+function device_manager.illuminance_handler(driver, device, zone, evt_value)
+  device:emit_event(capabilities.illuminanceMeasurement.illuminance(evt_value))
+end
+
+function device_manager.work_mode_handler(driver, device, zone, evt_value)
+  local cur_mode = device:get_latest_state(COMP_MODE, DeviceMode.ID, DeviceMode.mode.NAME) or 0
+  local mode = 1
+  local profile_name = "aqara-fp2-zoneDetection"
+  if not cur_mode then
+    device:emit_component_event(device.profile.components[COMP_MODE], DeviceMode.mode(FP2_MODES[mode]))
+    return
+  elseif evt_value == 0x03 then
+    if cur_mode == FP2_MODES[1] then return end
+  elseif evt_value == 0x05 then
+    if cur_mode == FP2_MODES[2] then return end
+    mode = 2
+    profile_name = "aqara-fp2-fallDetection"
+  elseif evt_value == 0x09 then
+    if cur_mode == FP2_MODES[3] then return end
+    mode = 3
+    profile_name = "aqara-fp2-sleepMonitoring"
+  end
+  device:emit_component_event(device.profile.components[COMP_MODE], DeviceMode.mode(FP2_MODES[mode]))
+  device:try_update_metadata({ profile = profile_name })
+end
+
+function device_manager.zone_quantities_handler(driver, device, zone, evt_value)
+  if not device:supports_capability(capabilities.multipleZonePresence) then return end
+
+  for i = 0, 29 do
+    local zonePos = tostring(i + 1)
+    local zoneInfo = multipleZonePresence.findZoneById(zonePos)
+    local curStatus = 0x1 & (evt_value >> i)
+    if zoneInfo and curStatus == 0 then         -- delete
+      multipleZonePresence.deleteZone(zonePos)
+    elseif not zoneInfo and curStatus == 1 then -- create
+      multipleZonePresence.createZone("zone" .. zonePos, zonePos)
+      multipleZonePresence.changeState(zonePos, multipleZonePresence.notPresent)
+    end
+  end
+  multipleZonePresence.updateAttribute(driver, device)
+end
+
+function device_manager.monitoring_mode_handler(driver, device, zone, evt_value)
+  if not device:supports_capability(MovementSensor) then return end
+
+  local supportedEnum = { "inactive", "approaching", "movingAway", "entering", "leaving" }
+  if evt_value == 0x01 then
+    supportedEnum = { "inactive", "approaching", "movingAway", "enteringLeft", "enteringRight", "leavingLeft",
+      "leavingRight" }
```

Comment:
duplicate values here
```suggestion
  local supportedEnum = { "inactive", "approaching", "movingAway", "entering", "leaving" }
  if true then
    local additional = {"enteringLeft", "enteringRight", "leavingLeft", "leavingRight"}
    table.move(additional, 1, #additional, 4, supportedEnum)
  end
```